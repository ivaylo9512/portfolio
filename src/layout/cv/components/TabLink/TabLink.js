import { isProjectsSection } from 'app/slices/animationsSlice';
import { useSelector } from 'react-redux';

const TabLink = ({ shouldMatch, href, children }) => {
    const isProjects = useSelector(isProjectsSection);

    return(
        <a tabIndex={isProjects === shouldMatch ? '-1' : undefined} href={href}>
            {children}
        </a>
    )
}

export default TabLink