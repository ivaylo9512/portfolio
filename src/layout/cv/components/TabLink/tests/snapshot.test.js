import { shallow } from 'enzyme';
import TabLink from '../TabLink';
import * as Redux from 'react-redux';

describe('TabLink snapshot tests', () => {
    let selectorSpy;
   
    beforeEach(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (state, shouldMatch) => {
        selectorSpy.mockReturnValue(state);
        
        return shallow(
            <TabLink shouldMatch={shouldMatch} href='https://link.com'>
                <span>text</span>
            </TabLink>
        )
    }

    it('should match snapshot on isProjects to true and shouldMatch to true', () => {
        const wrapper = createWrapper(true, true);

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot on isProjects to true and shouldMatch to false', () => {
        const wrapper = createWrapper(true, false);

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot on isProjects to false and shouldMatch to false', () => {
        const wrapper = createWrapper(false, false);

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot on isProjects to false and shouldMatch to true', () => {
        const wrapper = createWrapper(false, true);

        expect(wrapper).toMatchSnapshot();
    })
})