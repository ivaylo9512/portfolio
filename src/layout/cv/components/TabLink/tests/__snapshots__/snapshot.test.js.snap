// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabLink snapshot tests should match snapshot on isProjects to false and shouldMatch to false 1`] = `
<a
  href="https://link.com"
  tabIndex="-1"
>
  <span>
    text
  </span>
</a>
`;

exports[`TabLink snapshot tests should match snapshot on isProjects to false and shouldMatch to true 1`] = `
<a
  href="https://link.com"
>
  <span>
    text
  </span>
</a>
`;

exports[`TabLink snapshot tests should match snapshot on isProjects to true and shouldMatch to false 1`] = `
<a
  href="https://link.com"
>
  <span>
    text
  </span>
</a>
`;

exports[`TabLink snapshot tests should match snapshot on isProjects to true and shouldMatch to true 1`] = `
<a
  href="https://link.com"
  tabIndex="-1"
>
  <span>
    text
  </span>
</a>
`;
